pkg/controller/kogitoapp/resource/deployment_config.go

```diff
@@ -194,54 +203,72 @@ func setReplicas(kogitoApp *v1alpha1.KogitoApp, dc *appsv1.DeploymentConfig) {
 	dc.Spec.Replicas = replicas
 }
 
-// SetInfinispanEnvVars sets Infinispan variables to the given KogitoApp instance DeploymentConfig by reading information from the KogitoInfra
-func SetInfinispanEnvVars(cli *client.Client, kogitoInfra *v1alpha1.KogitoInfra, kogitoApp *v1alpha1.KogitoApp, dc *appsv1.DeploymentConfig) error {
-	if dc != nil && kogitoApp != nil &&
+// CreateInfinispanProperties creates Infinispan properties by reading information from the KogitoInfra
+func CreateInfinispanProperties(cli *client.Client, kogitoInfra *v1alpha1.KogitoInfra, kogitoApp *v1alpha1.KogitoApp) (envs []corev1.EnvVar, appProps map[string]string, err error) {
```

Comment:
Please coordinate these changes with https://github.com/kiegroup/kogito-cloud-operator/pull/326
The `CreateInfinispanProperties` and `CreateKafkaProperties` functions don't have to be in `deployment_config.go ` as they don't have anything to do with DeploymentConfig any more, can be transferred to infrastructure package.

Reply:
These two functions are specialized for kogito app only. They are supposed to be deprecated as part of https://issues.redhat.com/browse/KOGITO-1998 . And there is already generic code for creating infinispan and kafka properties in the `infrastructure` package. So I moved the two functions to two new files and kept them in the kogito app package. Please take a look. Thanks.

Reply:
Will recheck once the https://github.com/kiegroup/kogito-cloud-operator/pull/326 is merged and this PR is rebased.